File: 1207-Unique-Number-of-Occurrences/main.go
package main

func uniqueOccurrences(arr []int) bool {

    freq := make(map[int]int)

    for _, num := range arr {
        freq[num] += 1
    }

    occurrence := make(map[int]bool)

    for _, count := range freq {
        if _, ok := occurrence[count]; ok {
            return false
        }

        occurrence[count] = true
    }

    return true
}

File: 1207-Unique-Number-of-Occurrences/main.rs
use std::collections::HashMap;
use std::collections::HashSet;

impl Solution {
    pub fn unique_occurrences(arr: Vec<i32>) -> bool {
        
        let mut freq: HashMap<i32,i32 > = HashMap::new();

        for &num in &arr {
            *freq.entry(num).or_insert(0) += 1;
        }

        let mut occurrence = HashSet::new();
        for &count in freq.values(){
            if !occurrence.insert(count) {
                return false;
            }
        }
        true
    }
}
